src/read.jl

```diff
@@ -196,43 +196,35 @@ function open_or_read_xlsx(source::Union{IO, AbstractString}, read_files::Bool,
     xf = XLSXFile(source, enable_cache, read_as_template)
 
     try
-        for f in xf.io.files
+        for (i, f) in enumerate(ZipArchives.zip_names(xf.io))
 
```

Comment:
```suggestion
        for i in 1:ZipArchives.zip_nentries(xf.io)
            f = ZipArchives.zip_name(xf.io, i)
```
This should be faster because it avoids allocating all of the entry names at once.